cnd/src/storage.rs

```diff
@@ -692,8 +1182,79 @@ impl Load<herc20::Params> for Storage {
     }
 }
 
-// TODO: inserting ethereum redeem and lightning refund from other party is
-// common to all redeemers on lightning. Could extract that into a function
+#[async_trait::async_trait]
+impl Load<hbit::Params> for Storage {
+    async fn load(&self, id: LocalSwapId) -> anyhow::Result<hbit::Params> {
+        use crate::db::schema::swaps;
+
+        let (swap, hbit, secret_hash) = self
+            .db
+            .do_in_transaction::<_, _, anyhow::Error>(move |conn| {
+                let key = Text(id);
+
+                let swap: Swap = swaps::table
+                    .filter(swaps::local_swap_id.eq(key))
+                    .first(conn)?;
+
+                let hbit: Hbit = Hbit::belonging_to(&swap).first::<Hbit>(conn)?;
+                let secret_hash = SecretHash::belonging_to(&swap)
+                    .first::<SecretHash>(conn)
+                    .optional()?;
+
+                Ok((swap, hbit, secret_hash))
+            })
+            .await
+            .context(db::Error::SwapNotFound)?;
+
+        let role = swap.role.0;
+        let secret_hash = match role {
+            Role::Alice => comit::SecretHash::new(self.seed.derive_swap_seed(id).derive_secret()),
+            Role::Bob => secret_hash.ok_or_else(|| NoSecretHash(id))?.secret_hash.0,
+        };
+
+        let params = build_hbit_params(self.seed, hbit, id, role, secret_hash)?;
+
+        Ok(params)
+    }
+}
+
+#[async_trait::async_trait]
+impl Load<identity::Bitcoin> for Storage {
```

Comment:
I can see why you are doing this, it is actually in-line with how we designed the `Storage` abstraction :sweat_smile: 

The thing I find weird about this is that the caller actually has all the data to make these decisions so I think it would still probably do this outside and not load the data that we persist 2 lines before we call this fn in the controller? :man_shrugging: 

We also have an additional error case here that is one of these "should never happen" ..